docs/source/notes/amp_examples.rst

```diff
@@ -203,8 +223,139 @@ after all optimizers used this iteration have been stepped::
+
+    MyModel(nn.Module):
+        ...
+        @autocast()
+        def forward(self, input):
+           ...
+
+    # Alternatively
+    MyModel(nn.Module):
+        ...
+        def forward(self, input):
+            with autocast():
+                ...
+
+The following now autocasts in ``dp_model``'s side threads (which execute ``forward``) and the main thread
+(which executes ``loss_fn``)::
+
+    model = MyModel()
+    dp_model = nn.DataParallel(model)
+
+    with autocast():
+        output = dp_model(input)
+        loss = loss_fn(output)
+
+DistributedDataParallel, one GPU per process
+--------------------------------------------
+
+:class:`torch.nn.parallel.DistributedDataParallel`'s documentation recommends one GPU per process for best
+performance.  In this case, usages of :class:`autocast` and :class:`GradScaler` are not affected.
+No gotchas (side-thread-related or otherwise) apply.
+
+DistributedDataParallel, multiple GPUs per process
+--------------------------------------------------
+
+Here :class:`torch.nn.parallel.DistributedDataParallel` may spawn a side thread to run the forward pass on each
+device, like :class:`torch.nn.DataParallel`.  :ref:`The fix is the same<amp-dataparallel>`:
+apply autocast as part of your model's ``forward`` method to ensure it's enabled in side threads.
+
+.. _amp-custom-examples:
+
+Autocast and Custom Autograd Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your network uses :ref:`custom autograd functions<extending-autograd>`
+(subclasses of :class:`torch.autograd.Function`), :class:`autocast` may work out of the box,
+depending what internal operations your function(s) execute.
+However, you should double-check if they match one of the special cases below.
+
+Functions that should allow autocasting
+---------------------------------------
+
+The :func:`torch.cuda.amp.custom_fwd` and :func:`torch.cuda.amp.custom_bwd` decorators (with no arguments)
+ensure that ``forward`` executes with whatever autocast state surrounds the point of use, and that
+``backward`` executes with the same autocast state as ``forward`` (which can prevent type mismatch errors)::
+
+    class MyMM(torch.autograd.Function):
+        @staticmethod
+        @custom_fwd
+        def forward(ctx, a, b):
+            ctx.save_for_backward(a, b)
+            return a.mm(b)
+        @staticmethod
+        @custom_bwd
+        def backward(ctx, grad):
+            a, b = ctx.saved_tensors
+            return grad.mm(b.t()), a.t().mm(grad)
+
+Functions that need a particular datatype
+-----------------------------------------
+
+If you know your function requires a certain precision, or if it wraps a backend with limited datatype support,
+you may want it to disallow autocasting.
+
+Consider a custom autograd function wrapping
+`CUDA extensions <https://pytorch.org/tutorials/advanced/cpp_extension.html>`_
+that were only compiled for ``float32``.  You can force the function to run in ``float32`` at the point of use,
```

Comment:
The 'you' here is ambiguous. The first solution is for program authors to deal with this issue. The second solution is for function authors.